### PR TITLE
Resolves #826 - /tagnyan admin blocking

### LINKED ISSUE #826: /tagnyan admin unblocked
Continued from http://totalfreedom.boards.net/thread/28731/pretend-admin?page=1&scrollTo=317067

### PR DESCRIPTION
**Pull request by mayokek**
Tested and worked like normal. No errors.

### CODE REVIEW

File: src/me/StevenLawson/TotalFreedomMod/Commands/Command_tagnyan.java
Comment (JeromSar) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_tagnyan.java]: This should be case insensitive.

File: src/me/StevenLawson/TotalFreedomMod/Commands/Command_tagnyan.java
Comment (JeromSar) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_tagnyan.java]: The tag is set, whether this message is sent or not.

**Comment by JeromSar** [comment]
@GLaDOS12 Hello,

Thank you for contributing. Next to the issues mentioned above, this PR also lacks an associated issue (see contributing guidelines). Please fix all the issue with this PR so it may be merged.

**Comment by OxLemonxO** [comment]
What JeromSar means is, it doesn't actually prevent the tag nyan, even if it sends the message.

File: src/me/StevenLawson/TotalFreedomMod/Commands/Command_tagnyan.java
Comment (JeromSar) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_tagnyan.java]: Still doesn't actually block the tag.

File: src/me/StevenLawson/TotalFreedomMod/Commands/Command_tagnyan.java
Comment (JeromSar) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_tagnyan.java]: Tags may be longer than just one argument.

File: src/me/StevenLawson/TotalFreedomMod/Commands/Command_tagnyan.java
Comment (xDestroyer217) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_tagnyan.java]: Yep, just sends the message, doesn't block the issue.

**Comment by xDestroyer217** [comment]
I think this is on hold @JeromSar since it's been 20 days+

**Comment by JeromSar** [comment]
@xDestroyer217 That is not what "on hold" means for PRs.

**Comment by xDestroyer217** [comment]
@JeromSar I mean inactive, sorry.

File: src/me/StevenLawson/TotalFreedomMod/Commands/Command_tagnyan.java
Comment (Wild1145) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_tagnyan.java]: Has this even been tested? I cannot see how the hell that will work.

File: src/me/StevenLawson/TotalFreedomMod/Commands/Command_tagnyan.java
Comment (Wild1145) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_tagnyan.java]: Explains why there was a random ) on line 21...

File: src/me/StevenLawson/TotalFreedomMod/Commands/Command_tagnyan.java
Comment (Wild1145) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_tagnyan.java]: Can this not be pulled from another list? I am sure we already have defined a load of these sorts of blocked words??

**Comment by LifeAsStranger** [comment]
Will update soon

**Comment by LifeAsStranger** [comment]
Hello @JeromSar, I have updated to latest code of mine, I have tested on my LAN server and worked well. If there anything I need to change, just tell me. :+1:

File: src/me/StevenLawson/TotalFreedomMod/Commands/Command_tagnyan.java
Comment (Wild1145) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_tagnyan.java]: I would suggest telling them the word that they have used so "The tag you have set contains a forbidden word ( < WORD NAME > )

File: src/me/StevenLawson/TotalFreedomMod/Commands/Command_tagnyan.java
Comment (Wild1145) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_tagnyan.java]: I would suggest to either you or @JeromSar to create a removeTag option within the player data, just a bit cleaner.

Comment (LifeAsStranger) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_tagnyan.java]: That can be implement later when TFM 5.0 have been release.

**Comment by LifeAsStranger** [comment]
@Wild1145, I have updated it to make the TFM_PlayerData bit cleaner and add the forbidden word (WORD)

File: src/me/StevenLawson/TotalFreedomMod/Commands/Command_tagnyan.java
Comment (JeromSar) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_tagnyan.java]: This approach deletes a previous tag which may not have been blocked.

Comment (LifeAsStranger) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_tagnyan.java]: That is a problem that I am stuck on.

Comment (LifeAsStranger) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_tagnyan.java]: @JeromSar, that is the main problem.

Comment (JeromSar) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_tagnyan.java]: @LifeAsStranger No it isn't. The command should block a new tag, if it contains forbidden words, but it should never remove a pre-existing tag.

Comment (LifeAsStranger) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_tagnyan.java]: I mean, there isn't a code to remove the new one but leave the pre-existing tag untouchable.
Unless you use ``` event.setCancelled(true); ``` you will need to fix everything up.

Comment (JeromSar) [inline src/me/StevenLawson/TotalFreedomMod/Commands/Command_tagnyan.java]: @LifeAsStranger If you don't modify the tag until it's been verified as "safe", then it's possible. Everything is possible in java, since it is a [turing complete](https://en.wikipedia.org/wiki/Turing_completeness) programming language.

**Comment by AvalancheYT** [comment]
@JeromSar, after the long time. I decided to continue with this.
After some testing on LAN server, I finally got this completed. Please review this.

**Comment by AvalancheYT** [comment]
@JeromSar, CLA Signed.

**Review by Wild1145 — CHANGES_REQUESTED** [review]
Your PR currently has conflicts with the main branch that is stopping it from being able to be merged. Please check and address this before we can merge the PR.

**Comment by mayokek** [comment]
@Wild1145, it's already been added and I created another PR for this.

**Pull request closed without merging**
